src/redmodding.ts

```diff
@@ -60,6 +85,146 @@ export const REDmoddingTools = [
+  const tools = state.settings.gameMode.discovered[gameId]?.tools || {};
+  return Object.keys(tools).map((id) => tools[id])
+    .filter((iter) => (iter !== undefined) && (iter.path !== undefined))
+    .find((iter) => path.basename(iter.path).toLowerCase() === `redMod.exe`);
+};
+
+const autoDeployAction: V2077ActionFunc = (
+  vortexApi: VortexApi,
+  _features: Features,
+  _instanceIds: string[],
+): VortexActionResult => {
+  const state = vortexApi.store.getState();
+  const profile = selectors.activeProfile(state);
+  vortexApi.store.dispatch(setNeedToRun(profile.id, true));
+};
+
+export const wrapVortexActionFunc =
+  (
+    vortex: VortexExtensionContext,
+    vortexApiThing,
+    actions: V2077SettingsView,
+  ): VortexActionFunc =>
+    //
+    // This is the function that Vortex calls.
+    (instanceIds?: string[]): VortexActionResult => {
+      //
+      const vortexApi: VortexApi = { ...vortex.api, log: vortexApiThing.log };
+
+      return actions.actionOrCondition(
+        vortexApi,
+        CurrentFeatureSet,
+        instanceIds,
+      );
+    };
+
+const autoDeployActionCondition: V2077ActionConditionFunc = (
+  vortexApi: VortexApi,
+  features: Features,
+  _instanceIds: string[],
+): VortexActionConditionResult => {
+  if (!FeatureEnabled(features.REDmodding)) {
+    return false;
+  }
+  const state = vortexApi.store.getState();
+  const gameMode = selectors.activeGameId(state);
+  const tool = redModTool(state, gameMode);
+  return isSupported(gameMode) && (tool !== undefined);
+};
+
+export const wrapVortexActionConditionFunc =
+  (
+    vortex: VortexExtensionContext,
+    vortexApiThing,
+    actions: V2077SettingsView,
+  ): VortexActionConditionFunc =>
+    //
+    // This is the function that Vortex calls.
+    (instanceIds?: string[]): VortexActionConditionResult => {
+      //
+      const vortexApi: VortexApi = { ...vortex.api, log: vortexApiThing.log };
+
+      return actions.condition(
+        vortexApi,
+        CurrentFeatureSet,
+        instanceIds,
+      );
+    };
+
+const checkFailedResult = (
+  t: TranslationFunction,
+  gameMode: string,
+  reason: 'missing' | 'outdated',
+): types.ITestResult => {
+  const short = {
+    missing: t(`REDmod not installed`),
+  }[reason];
+
+  const long = {
+    missing: t(`meow`),
+  }[reason];
+
+  const res: types.ITestResult = {
+    severity: `warning`,
+    description: {
+      short,
+      long,
+      localize: false,
+    },
+  };
+  return res;
+};
+
+export const wrappedCheckFunc =
+(
+  vortex: VortexExtensionContext,
+  vortexApiThing,
+  settingsView: V2077SettingsView,
+): VortexCheckFunc =>
+  //
+  // This is the function that Vortex calls.
+  (): Promise<VortexCheckResult> => {
+    //
+    const vortexApi: VortexApi = { ...vortex.api, log: vortexApiThing.log };
+
+    return settingsView.test(
+      vortexApi,
+      CurrentFeatureSet,
+    );
+  };
+
+export const checkTest: V2077CheckFunc = async (
```

Comment:
I didn't quite follow this, is this for a user-initiated version of the DLC check that we do internally (below)?

Reply:
tbh, I mostly was implementing the same things that the fnis integration plugin was implementing knowing that it was implementing a settings page and interactions with a third party tool. I was not sure or everything that mattered or not for adding in the settings page UI specifically. 

But ya, I believe that might be the purpose of this function, and probably it could be removed.